test/service_manager/service_manager.go

```diff
@@ -109,12 +118,39 @@ func (c *ServiceManager) waitForBroker() {
 
 	select {
 	case <-timeout:
-		log.Fatal("Broker isn't ready within expected timeout")
+		log.Println("Broker isn't ready within expected timeout")
+		errObtainingLogs := c.executeCmdWithLogging(
+			"obtain logs from zookeeper and kafka containers",
+			"docker-compose",
+			"-f", pathToDockerComposeImage, "logs",
+			)
+		if errObtainingLogs != nil {
+			log.Println("Problem obtaining logs from kafka and zookeeper containers")
+			log.Fatal(errObtainingLogs.Error())
+		}
+		log.Fatal("containers are not in suitable state")
 	case <-brokerIsReadyChannel:
 		log.Println("Container (Broker) is ready")
 	}
 }
 
+func (c *ServiceManager) executeCmdWithLogging( commandDescription ,commandName string, commandArgs ...string) error{
+
+	cmd := exec.Command(commandName, commandArgs...)
+	var stdout, stderr bytes.Buffer
+	// redirect Stdout and Stderr of command to buffers
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
```

Comment:
AFAIU we are redirecting entire std out and err in two bytes buffer in memory, are we sure it's going to work? what about the max size of these buffers? I have doubt about this.